### PR TITLE
Improve test shell scripts

### PR DESCRIPTION
**Pull request by pabs3**
- Standardise on `set -eux` for all test shell scripts
- Run git-mediate via `$PATH` instead of a location stored in a variable
- Avoid passing options to echo
- Use `=` instead of `==` within `[ ]` tests
- Quote references to shell variables
- Use braces around references to shell variables
- Set the test user home directory to the test directories
- Wrap multi-command shell statements more
- Add shebangs to the shell scripts
- Consistently single-quote static strings containing spaces
- Ensure the tests run in a clean directory and clean it on exit
- Print the current directory in all tests for consistency
- Indicate successful completion of tests

### CODE REVIEW

**Review by yairchu — CHANGES_REQUESTED** [review]
Thanks a lot!
I've cherry-picked one of the commits (which is actually necessary for the tests to properly run on macOS) and for some of the other changes the tests are not working for me and I suggested a few changes.

File: test/config/test [hunk 1]
```diff
@@ -24,7 +24,7 @@ git config --unset merge.conflictstyle
 git config --global merge.conflictstyle diff2
 git-mediate -s
 style=$(git config merge.conflictstyle)
-if [ "$style" == "diff3" ]; then
+if [ "$style" = "diff3" ]; then
```
[the remark this answers is not in the document]
Comment (pabs3) [reply test/config/test]: Indeed, fixed.

File: test/config/test [hunk 2]
```diff
@@ -3,8 +3,11 @@ set -eux
 DIR="$( dirname "$0" )"
 cd "${DIR}"
 
+export HOME="${DIR}"
+
 git init
 
+git config --global merge.conflictstyle diff3
```
[the remark this answers is not in the document]
Comment (pabs3) [reply test/config/test:20]: Hmm, that happens on Linux too, fixed.

File: test/spaces/test [hunk 3]
```diff
@@ -19,6 +21,7 @@ git add 'a b'
 (git stash pop >/dev/null && exit 1) || echo "Conflict set up successfully"
 # Set up of conflict over
 
+git config --global merge.conflictstyle diff3
```
[the remark this answers is not in the document]
Comment (pabs3) [reply test/spaces/test]: Fixed

**Comment by pabs3** [comment]
Thanks for merging and looking into the spaces issue and NUL option.

-- 
bye,
pabs

https://bonedaddy.net/pabs3/

**Pull request merged**